_posts/2015-06-09-Decoupling-the-Mind-of-the-API-Designer.md

```diff
@@ -0,0 +1,60 @@
+---
+title: Decoupling the Mind of the API Designer
+excerpt: Fostering Creativity with Semantically Rich APIs
+layout: post
+date: 2015-06-10 00:00:00 +0200
+author: stephen
+published: true
+---
+
+I recently watched a presentation called [The Humane Representation of Thought](https://vimeo.com/115154289) by Brett Victor in which he starts out his talk with this quote from "Genius: The Life and Science of Richard Feynman."
+
+> Feynman said to Dyson, and Dyson agreed, that Einstein's great work had sprung from physical intuition and that when Einstein stopped creating it was because "he stopped thinking in concrete physical images and became a manipulator of equations."
+
+It was interesting that Einstein's tools for communicating the complexity of the universe were what began to hinder his creativity in exploring that complexity. His thought become intertwined with equations so much, it actually stifled his inspiration.
+
+It struck me that we're doing the same things in the world of API design. We're of course happy to have won the battle over technologies like SOAP and WSDL, but we fail to see our tools and methodologies are leading us right back to where we were. We have coupled our thought and creativity to these tools and methodologies and have shackled our imagination to specs and protocols.
+
+## Chewing the CRUD
+
+If you survey the modern API space, you'll see this coupled design pattern is the norm. The first step in this pattern is usually to carefully craft a URL that conveys to the users the meaning of the content. The next step is to pick the HTTP method the users will use in interacting with that URL. Our thought, design, and creativity are all directly coupled with the HTTP implementation in this process, and it pushes the question "What does my API do?" to later design steps.
+
+And what we haven't seen is this—we believe that thinking about URLs and HTTP methods *is* answering the question "What does my API do?" We define our resources, their CRUD actions, and place the burden on the user to glean the meaning and use of our API. Instead of giving users a universe to explore, we've giving them equations to manipulate.
+
+## What is Coupling?
+
+Before continuing, it's important to define what I mean when I refer to the term "coupling." It's used in many contexts and many industries to refer to the concept of two things being united or joined together. When I use coupling in the context of API design, I am speaking of joining together the design and the technology used to implement that design in such a way that it makes it difficult to separate the two.
```

Comment:
The big debate :) In American writing, the punctuation goes inside the quotation marks.